src/render/reporting.js

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable camelcase, no-cond-assign, compat/compat */
+
+/*
+ * @function
+ * @name hasRendered
+ * @description determine if a DOM element is visibile on a page
+ * @param {DOM element} el - a list of bidders (also referred to as interest groups)
+ * @return {boolean} whether or not an element is visible on the screen
+ */
+export const hasRendered = el => {
+	if (!(el instanceof Element)) { throw Error(`${el} is not a DOM element.`); }
+
+	const { display, opacity, visibility } = getComputedStyle(el);
+	if (display === 'none') { return false; }
+	if (visibility !== 'visible') { return false; }
+	if (opacity < 0.1) { return false; }
+
+	const { left, height, top, width } = el.getBoundingClientRect();
+	const { offsetHeight } = el;
+	const { offsetWidth } = el;
+	if (offsetWidth + offsetHeight + height + width === 0) { return false; }
+
+	const x = left + offsetWidth / 2;
+	if (x < 0) { return false; }
+	if (x > (document.documentElement.clientWidth || window.innerWidth)) { return false; }
+	const y = top + offsetHeight / 2;
+	if (y < 0) { return false; }
+	if (y > (document.documentElement.clientHeight || window.innerHeight)) { return false; }
+
+	let pointContainer = document.elementFromPoint(x, y);
+	do {
+		if (pointContainer === el) { return true; }
+	} while (pointContainer = pointContainer.parentNode);
+
+	return false;
+};
+
+/*
+ * @function
+ * @name getSellerReport
+ * @description given the results of an auction, grab the report from the seller
+ * @author Newton <cnewton@magnite.com>
+ * @param {object} conf - an auction configuration object
+ * @param {object} results - the results of the auction
+ * @return {object} an object of data to pass back to the buyers report
+ */
+export const getSellerReport = async (conf, results) => {
+	const { report_result } = await import(conf.decision_logic_url);
+
+	// check if there is even a function
+	if (!report_result && typeof report_result !== 'function') {
+		return null;
+	}
+
+	// generate a report by providing all of the necessary information
+	return report_result(conf, {
```

Comment:
ideally we would have an `await` and a `try`/`catch` here as the report function will require an async call to the sellers server to report.

Reply:
ok I get why you don't need the await (since you are returning a promise here) but you'll need the try/catch

Reply:
and since you'll be adding a try/catch you'll need the await, ha ha

Reply:
This works the same as the bidding utility function. It's a dynamic import so no need for that. If you look at the demo it works and calls the function and returns data in the console.

Reply:
fixed